### DISCUSSION TITLE
V3 FormObject Real-time Validation

### DISCUSSION THREAD

**Discussion by marksalmon** [livewire/livewire#5906] [1. Help]
I am struggling with real-time validation and I wonder if someone can explain where I'm going wrong. 

The docs state:

> Livewire handles this sort of thing automatically. By using .live or .blur on wire:model, Livewire will send network requests as the user fills out the form. Each of those network requests will run the appropriate validation rules before updating each property. If validation fails, the property won't be updated on the server and a validation message will be shown to the user:

I am finding that although the validation rules are being correctly evaluated and displaying the error, the value **is** being updated server-side.

I can reproduce with this simple example:

app/Livewire/TestComponent
```php
<?php

namespace App\Livewire;

use App\Livewire\Forms\TestForm;
use Livewire\Component;

class TestComponent extends Component
{
    public TestForm $form;

    public function render()
    {
        return <<<'HTML'
        <div class="space-y-8">
            <div>@dump($form)</div>

            <div>@error('form.input') <span class="error">{{ $message }}</span> @enderror</div>

            <div><input wire:model.blur="form.input" /></div>
        </div>
        HTML;
    }
}
```

app/Livewire/Forms/TestForm.php
```php
<?php

namespace App\Livewire\Forms;

use Livewire\Attributes\Rule;
use Livewire\Form;

class TestForm extends Form
{
    #[Rule('string|min:6')]
    public $input;
}
```

<img width="779" alt="image" src="https://github.com/livewire/livewire/assets/7000886/bc1b436c-5b92-47d1-9e25-0868e53c86c1">

You can see in the dumped TestForm object, that the ```$input``` has indeed been updated.

**Comment by marksalmon** [comment]
## UPDATE

It is not related to being a form object. the same thing happens if the property belongs directly to the component:

app/LivewireTestComponent
```php
<?php

namespace App\Livewire;

use Livewire\Attributes\Rule;
use Livewire\Component;

class TestComponent extends Component
{
    #[Rule('string|min:6')]
    public $componentInput;

    public function render()
    {
        return <<<'HTML'
        <div class="space-y-8">
            <div>@dump($componentInput)</div>

            <div>@error('componentInput') <span class="error">{{ $message }}</span> @enderror</div>

            <div><input wire:model.blur="componentInput" /></div>
        </div>
        HTML;
    }
}

```

<img width="841" alt="image" src="https://github.com/livewire/livewire/assets/7000886/6b1f1326-6c65-4a6a-822d-d30638ef0114">

**Comment by joshhanley** [comment]
@marksalmon thanks for reporting! This looks like a similar issue as #5782 so let's continue the discussion there.
